### PR TITLE
[4.0] Template Atum - Cleanup and fix Logo Usage

### PR DESCRIPTION
**Pull request by coolcat-creations**
Pull Request for Issue #28826

### Summary of Changes
- Removed unused logos
- Renamed logos to more generic names
- Moved logos to subfolder
- Renamed variables to more recognizeable names
- Changed scss
- Added sizes to svg instead scss
- Changed order of Login input fields 
- Changed Language Strings

### Testing Instructions
- Install the patch
- npm run build:css
- check login logo, logo in headerbar, expand and collapse the sidebar
- upload own logos:
Custom header collapsed Logo recommended max. size 25px x 25px
Custom header expanded Logo recommended max. size 275px x 45 px

- Question: The Logo Variables repeat all over the template files, would it be an idea to have a logos.php file and include it into the severall template files?

- Login Logo appears black for now but only until the PR #28788 will be merged

grafik

grafik

grafik

grafik
grafik
grafik

### CODE REVIEW

**Comment by brianteeman** [comment]
what is the reasoning behind changing the name from site logo to header logo?

**Comment by coolcat-creations** [comment]
> what is the reasoning behind changing the name from site logo to header logo?

@brianteeman IMO it's not the "site" but the backend or administration (Maybe lost in translation?)
Additionally it's the logo that is displayed in the header - there is another logo displayed at login - so the login logo could be also the site logo and then the confusion with new users starts.

I tried to find names where the user knows exactly where the logos might appear. 

What do you think about it?

File: administrator/templates/atum/templateDetails.xml [hunk 1]
```diff
@@ -111,35 +111,35 @@
 
 			<fieldset name="images" label="TPL_ATUM_IMAGE_SETTINGS_LABEL">
 				<field
-					name="siteLogo"
+						name="loginLogo"
```
Comment (Quy) [inline administrator/templates/atum/templateDetails.xml:114]: Remove indentation.

**Comment by brianteeman** [comment]
@coolcat-creations that doesnt convince me of the need to change its name

**Comment by coolcat-creations** [comment]
@brianteeman I don't understand why it does not convice you, what are the advantages of the label "Site Logo" if it's even not on the site but in the administration ?
We use everywhere in the backend the context **site** for the _frontend_ and **administrator** for _backend_. I think naming it **site logo** is quite confusing and not consistent.

**Comment by brianteeman** [comment]
i am not in favour of a name that is tied to a location. Your argument for changing it from site is valid though. how about "brand"? then it means either joomla brnad logo or as often used the agency brand

**Comment by coolcat-creations** [comment]
@brianteeman brand is fine too. 
Brand collapsed and Brand expanded 
or 
Brand large and Brand small ?

**Comment by brianteeman** [comment]
> @brianteeman brand is fine too.
> Brand collapsed and Brand expanded
> or
> Brand large and Brand small ?

Large and Small

**Comment by coolcat-creations** [comment]
@brianteeman - Done :-)

**Comment by brianteeman** [comment]
I answered the question before looking at the code ;)

File: administrator/templates/atum/error_full.php [hunk 2]
```diff
@@ -85,8 +84,8 @@
 			<div class="d-flex align-items-center">
 				<a class="logo" href="<?php echo Route::_('index.php'); ?>"
 				   aria-label="<?php echo Text::_('TPL_ATUM_BACK_TO_CONTROL_PANEL'); ?>">
-					<img src="<?php echo $siteLogo; ?>" alt="">
-					<img class="logo-small" src="<?php echo $smallLogo; ?>" alt="">
+					<img src="<?php echo $logoBrandLarge; ?>" alt="">
+					<img class="logo-collapsed" src="<?php echo $logoBrandSmall; ?>" alt="">
```
Comment (Quy) [inline administrator/templates/atum/error_full.php]:
```suggestion
					<img src="<?php echo $logoBrandLarge; ?>" alt="<?php echo $logoBrandLargeAlt; ?>">
					<img class="logo-collapsed" src="<?php echo $logoBrandSmall; ?>" alt="<?php echo $logoBrandSmallAlt; ?>">
```

File: administrator/templates/atum/index.php [hunk 3]
```diff
@@ -94,14 +93,14 @@
...
+					<img src="<?php echo $logoBrandLarge; ?>" alt="<?php echo $logoBrandLargeAlt; ?>">
+					<img class="logo-collapsed" src="<?php echo $logoBrandSmall; ?>" alt="<?php echo $logoBrandSmallAlt; ?>">
 					</div>
 				<?php else : ?>
 					<a class="logo" href="<?php echo Route::_('index.php'); ?>"
 						aria-label="<?php echo Text::_('TPL_ATUM_BACK_TO_CONTROL_PANEL'); ?>">
-						<img src="<?php echo $siteLogo; ?>" alt="">
-						<img class="logo-small" src="<?php echo $smallLogo; ?>" alt="">
+						<img src="<?php echo $logoBrandLarge; ?>" alt="">
+						<img class="logo-collapsed" src="<?php echo $logoBrandSmall; ?>" alt="">
```
Comment (Quy) [inline administrator/templates/atum/index.php]:
```suggestion
						<img src="<?php echo $logoBrandLarge; ?>" alt="<?php echo $logoBrandLargeAlt; ?>">
						<img class="logo-collapsed" src="<?php echo $logoBrandSmall; ?>" alt="<?php echo $logoBrandSmallAlt; ?>">
```

File: administrator/templates/atum/templateDetails.xml [hunk 4]
```diff
@@ -111,35 +111,35 @@
...
+						type="media"
+						label="TPL_ATUM_LOGIN_LOGO_LABEL"
```
Comment (Quy) [inline administrator/templates/atum/templateDetails.xml]:
```suggestion
					name="loginLogo"
					type="media"
					label="TPL_ATUM_LOGIN_LOGO_LABEL"
```

File: administrator/templates/atum/templateDetails.xml [hunk 5]
```diff
@@ -111,35 +111,35 @@
...
+				/>
+				<field
+						name="altLoginLogo"
+						type="text"
+						label="TPL_ATUM_ALTTEXT_LOGIN_LOGO_LABEL"
```
Comment (Quy) [inline administrator/templates/atum/templateDetails.xml]:
```suggestion
					name="altLoginLogo"
					type="text"
					label="TPL_ATUM_ALTTEXT_LOGIN_LOGO_LABEL"
```

**Comment by wilsonge** [comment]
If you can fix conflicts @coolcat-creations I'm happy to get this merged as seems very sensible!

**Comment by coolcat-creations** [comment]
@wilsonge I can't resolve conflicts.
grafik
in this PR logo-blue would be moved and renamed

**Comment by Quy** [comment]
Please sync branch. I am getting the following notice.

> PHP Notice:  Undefined index: home in \administrator\components\com_templates\src\Table\StyleTable.php on line 58

File: administrator/templates/atum/error_login.php [hunk 6]
```diff
@@ -29,20 +29,19 @@
...
-$joomlaLogo = $this->baseurl . '/templates/' . $this->template . '/images/logo.svg';
 
 require_once __DIR__ . '/Service/HTML/Atum.php';
 
 // Template params
-$siteLogo  = $this->params->get('siteLogo')
-	? Uri::root() . $this->params->get('siteLogo')
+$logoBrandLarge  = $this->params->get('logoBrandLarge')
+	? Uri::root() . $this->params->get('logoBrandLarge')
 	: $this->baseurl . '/templates/' . $this->template . '/images/logo-joomla-blue.svg';
```
Comment (infograf768) [inline administrator/templates/atum/error_login.php]: Are'nt you deleting logo-joomla-blue.svg i the PR?

File: administrator/templates/atum/error_login.php [hunk 7]
```diff
@@ -29,20 +29,19 @@
...
-$smallLogo = $this->params->get('smallLogo')
-	? Uri::root() . $this->params->get('smallLogo')
+$logoBrandSmall = $this->params->get('logoBrandSmall')
+	? Uri::root() . $this->params->get('logoBrandSmall')
 	: $this->baseurl . '/templates/' . $this->template . '/images/logo-blue.svg';
```
Comment (infograf768) [inline administrator/templates/atum/error_login.php]: Are'nt you deleting logo-blue.svg in the PR?

**Comment by coolcat-creations** [comment]
Can anyone give me a hint how to resolve conflicts here? It's greyed out...

**Comment by richard67** [comment]
@coolcat-creations Yes, if it becomes too complicated for the GitHub UI you can only do it in a local git client. I did it for you as good as I knew, but there is one thing you should check: If the SVG files added by your PR somehow were based on those deleted by this PR (i.e. a rename with move would have been better), you might have to update your new SVGs by the changes having done on the old SVGs meanwhile in the 4.0-dev branch.

**Comment by infograf768** [comment]
@coolcat-creations 
I still don't understand. You marked the conversation as resolved but:
You are deleting  `logo-joomla-blue.svg`
but it is still present in error_login.php
` 	: $this->baseurl . '/templates/' . $this->template . '/images/logo-joomla-blue.svg'; `
Same for `logo-blue.svg`

I have marked both conversations to unresolved until we get an explanation.

**Comment by coolcat-creations** [comment]
> @coolcat-creations
> I still don't understand. You marked the conversation as resolved but:
> You are deleting `logo-joomla-blue.svg`
> but it is still present in error_login.php
> ` : $this->baseurl . '/templates/' . $this->template . '/images/logo-joomla-blue.svg';`
> Same for `logo-blue.svg`
> 
> I have marked both conversations to unresolved until we get an explanation.

Thank you, sorry - I don't remember - I think I had overseen it. I corrected it now. Anyway it would make sense in another PR to centralize the logo usage instead of repeating in each document?

**Comment by coolcat-creations** [comment]
> @coolcat-creations Yes, if it becomes too complicated for the GitHub UI you can only do it in a local git client. I did it for you as good as I knew, but there is one thing you should check: If the SVG files added by your PR somehow were based on those deleted by this PR (i.e. a rename with move would have been better), you might have to update your new SVGs by the changes having done on the old SVGs meanwhile in the 4.0-dev branch.

Thank you very much... It was not complicated but It was greyed out and I had no permission (I don't know why) :)

**Comment by infograf768** [comment]
Hmm, looks like I have some issues here. 
Without modifying template style params:

Screen Shot 2020-07-21 at 18 32 51
Screen Shot 2020-07-21 at 18 32 17
Screen Shot 2020-07-21 at 18 32 00

If I modify the params for brand (using an image from the site images folder as we can't use an image from the template image folder), I get also whwn saving AND staying in the edit page

Screen Shot 2020-07-21 at 18 42 04

After closing page
Screen Shot 2020-07-21 at 18 43 23

**Comment by infograf768** [comment]
Looks like working (forgot to use npm in a first try)...

**Comment by coolcat-creations** [comment]
thank you

File: administrator/templates/atum/scss/blocks/_header.scss [hunk 8]
```diff
@@ -31,22 +31,26 @@
...
   .logo {
-    display: inline-block;
+    display: flex;
+    height: 100%;
     width: $sidebar-width;
-    transition: width .3s ease;
+    flex-flow: row nowrap;
+    align-items: center;
+    transition: all 0.3s ease-in-out;
+
```
Comment (Quy) [inline administrator/templates/atum/scss/blocks/_header.scss:40]:
```suggestion
```

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on d95085ece9076c260198b1b4b4c457cd099af8bb<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/28839">issues.joomla.org/tracker/joomla-cms/28839</a>.</sub>

**Comment by infograf768** [comment]
I have tested this item :white_check_mark: successfully on d95085ece9076c260198b1b4b4c457cd099af8bb<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/28839">issues.joomla.org/tracker/joomla-cms/28839</a>.</sub>

**Comment by infograf768** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/28839">issues.joomla.org/tracker/joomla-cms/28839</a>.</sub>

**Comment by infograf768** [comment]
Tks.

**Comment by coolcat-creations** [comment]
thank you!

**Pull request merged**